src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -6164,9 +6171,12 @@ and TcExprSequential (cenv: cenv) overallTy env tpenv (synExpr, _sp, dir, synExp
 
 and TcExprSequentialOrImplicitYield (cenv: cenv) overallTy env tpenv (sp, synExpr1, synExpr2, otherExpr, m) =
 
-    let isStmt, expr1, tpenv =
+    let isStmt, expr1, expr1Ty, tpenv =
         let env1 = { env with eIsControlFlow = (match sp with DebugPointAtSequential.SuppressNeither | DebugPointAtSequential.SuppressExpr -> true | _ -> false) }
-        TryTcStmt cenv env1 tpenv synExpr1
+        let expr, ty, tpenv = TcExprOfUnknownType cenv env1 tpenv synExpr1
```

Comment:
This just inlines the contents of `TryTcStmt`, doesn't it?
I would suggest either add the additional `expr1Ty` return value to `TryTcStmt` directly.

Or, of there are 0/1 occurences after this PR, remove it alltogether.

Reply:
i added the return-value to TryTcStmt (didn't know it was so rarely used)